edward/util/random_variables.py

```diff
@@ -579,9 +581,15 @@ def get_variables(x, collection=None):
 
   # Traverse the graph. Add each node to the set if it's in the collection.
   output = set([])
+  visited = set()
   nodes = set([x])
   while nodes:
     node = nodes.pop()
+
+    if node in visited:
+      continue
+    visited.add(node)
+
```

Comment:
A clear and simple fix. Should this also be done in `get_parents`, `get_children`, etc.?

Also I like that `set()` works over `set([])`. Could you replace the similar lines such as `output` to also do this?